src/FreeformIdeas.js

```diff
@@ -33,26 +43,35 @@ const FreeForm = ({ hideSourceOnDrag }) => {
             },
         }));
     };
+    const editIdea = useCallback((title) => {
+        setIdeas(update(ideas, {
+            [storedID]:{
+                $merge:{title},
+            },
+        }));
+    }, [ideas, storedID]);
     const addIdea = useCallback((title) => {
         const id = uuidv4()
-        const top = 200;
-        const left = 200;
+        const left = coords[0];
+        const top = coords[1];
         setIdeas(update(ideas, {[id]: {$set: {top,left,title}}}));
-    }, [ideas]);
+    }, [coords, ideas]);
     const handleClick = useCallback((e) => {
         if(e.target.className !== "FreeformMap") return;
         e.preventDefault();
+        setCoords([coords[0] = e.nativeEvent.layerX, coords[1] = e.nativeEvent.layerY]);
```

Comment:
```suggestion
        setCoords([e.nativeEvent.layerX, e.nativeEvent.layerY]);
```

You should never use the assignment operator inside an expression. It's confusing and can lead to subtle bugs like the one you've introduced here. You're mutating `coords` which is a no-no. In fact, you don't care about the old coords at all.

Reply:
Also, I'll let it slide for this since it's small, but the coords change should really have been its own PR.

Reply:
Got it. It should be able to just be an array, and it will replace the original coords entirely?
I'll try to not submit it in that way.

Reply:
`setCoords` completely replaces the state. `update` is nice to use when you're just replacing one or two things in a complex structure, but in this case it's entirely new data.